### DISCUSSION TITLE
Custom Component - Interative Dataframe similar to aggrid

### DISCUSSION THREAD

**Discussion by anubhavrohatgi** [writer/writer-framework#219] [General]
Hi,
I am interested in designing an interactive component similar to Aggrid for dataframes, where I want some buttons in rows which perform some actions on the rows. Also I would like to edit the data.
1.  How should I go about this?
2. In the bubble button example, how are you defining the event variable like flagged_message_id? From where is this varible formed, How will JS know this is the variable that will be carrying the data?

**Comment by FabienArcellier** [comment]
Hi @anubhavrohatgi,

That's a great idea. A prototype has already been shared on the issue #149. You should take a look.

You can take a look on Core Component as `ui/src/core_components/CoreText.vue` and `ui/src/core_components/input/CoreTextInput.vue` to have a concrete example on how to write a binding and how to manage events. The mapping is done directly by streamsync engine. You don't have to manage anything on python side to build your own component.

**Comment by NanisTe** [comment]
Hi @FabienArcellier 

is there already a solution in place or a different issue for this issue being closed?
I am investigating the use of streamsync but have to understand the possibilities it gives us.
